src/software/simulation/physics/physics_world.h

```diff
@@ -34,13 +40,131 @@ class PhysicsWorld
     PhysicsWorld(const PhysicsWorld&)            = delete;
 
     /**
-     * Returns the current state of the world
-     * @return the current state of the world
+     * Returns the field in the physics world
+     *
+     * @return the field in the physics world
+     */
+    const Field getField() const;
+
+    /**
+     * Returns the state of the ball in the physics world, if it exists
+     *
+     * @return the state of the ball in the physics world, if it exists
+     */
+    const std::optional<BallState> getBallState() const;
+
+    /**
+     * Returns the states and IDs of all yellow robots in the physics world.
+     *
+     * There is no guarantee as to what order the robots are returned in.
+     *
+     * @return the states and IDs of all yellow robots in the physics world
+     */
+    const std::vector<RobotStateWithId> getYellowRobotStates() const;
```

Comment:
Can we define a teamcolour enum and pass that in? There's a lot of duplicated code here

Reply:
After in-person discussion, we decided to leave the interface as two functions, but unify duplicate code into a helper function underneath.

Reply:
done